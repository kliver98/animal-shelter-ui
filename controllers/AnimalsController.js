import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: `animals`,
            data: animal,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
    getAnimal(name) {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
    updateAnimal(name) {
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    }
}

