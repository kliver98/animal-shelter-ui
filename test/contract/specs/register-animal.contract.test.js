import { provider } from '../config/init-pact';
import { AnimalController } from '../../../controllers';
import { Matchers } from '@pact-foundation/pact';

const animal = {
    'name': 'pruebaaa',
    'breed': 'Birmano',
    'gender': 'Male',
    'isVaccinated': true,
    'vaccines': ["gripe","peritonitisInfecciosa"]
}

describe('Given an animal service', () => {
    beforeAll(async() => {
        await provider.setup();
    });

    describe('When a request to create an animal is made', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'backend service is up',
                uponReceiving: 'a request to create an animal',
                withRequest: {
                    method: 'POST',
                    path: '/animals'
                },
                willRespondWith: {
                    status: 201,
                    body: Matchers.eachLike({
                        name: Matchers.string('Manchas'),
                        breed: Matchers.like("Bengali"),
                        //gender: Matchers.term({generate: 'Female', matcher: 'Female|Male'}),
                        gender: Matchers.like("Female"),
                        vaccinated: Matchers.boolean(true),
                        vaccines: Matchers.eachLike("Rabia", {min: 1})
                    }, {min: 1})
                }
            });
        });

        test('Then it should return the right data', async() => {
            const response = await AnimalController.register(animal);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});
