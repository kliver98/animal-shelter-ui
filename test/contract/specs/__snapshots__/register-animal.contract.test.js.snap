// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an animal service When a request to create an animal is made Then it should return the right data 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "name": "Manchas",
    "vaccinated": true,
    "vaccines": Array [
      "Rabia",
    ],
  },
]
`;
